Add Monstrosity Handling

tw-blob opened the pull request:
This should address #476

msprijatelj commented:
Agreed, what Bolts said, or written permission from Massif Press to use it in this way.

tw-blob commented:
I reached out to them, but haven't heard back yet. Comp/Con didn't have the structure table so I assumed the LCP didn't include the data. But I'll admit I didn't look to closely.

msprijatelj commented:
It’s hardcoded into the Reference section of the Compendium: https://compcon.app/#/compendium/reference

BoltsJ commented:
Yeah, it's in the description of the trait, so if it needs to be parsed, that's where you would pull it from.  That said, I think that getting permission is the better option here.

tw-blob commented on src/module/flows/structure.ts:
```diff
@@ -136,6 +147,46 @@ function structTableDescriptions(roll: number, remStruct: number): string {
+function hasMonstrosityClass(actor: LancerActor): boolean {
+  return (
+    actor.is_npc() &&
+    actor.items.some(
+      i =>
+        i.type === "npc_class" &&
+        i.system &&
+        typeof i.system === "object" &&
+        (i.system as any).lid === "npcc_monstrosity"
+    )
```
If the trait has the table and rules text it would make more sense to check for that, no?
Then you can remove it to let the Monstrosity roll on the normal table, or conversely add it to to other NPCs to roll on this structure table.

BoltsJ replied:
Yeah, true. Checking for the trait would be on the the `npc_feature` itemTypes.

BoltsJ commented on src/module/flows/structure.ts:
```diff
@@ -136,6 +147,46 @@ function structTableDescriptions(roll: number, remStruct: number): string {
+  switch (roll) {
+    // Multiple 1s
+    case 0:
+      return "The Monstrosity is destroyed.";
```
It would be nice to move it if you're up to it, but I'm not going to require it for this PR.

tw-blob commented:
The Monstrosity Structure Table is included with permission by Massif Press
![XnU9MetIcM](https://github.com/user-attachments/assets/d78fb910-7d58-44f8-bebe-f3c2d7385325)



I've removed the secondary structure rolls for Monstrosities and added the dismemberment prompt and damage roll. The Hull check is untouched since both mechs and Monstrosities follow the same logic there.

Currently the damage roll is included in the structure flow. I'm unsure if there is a better way to handle it or a preferred place to include it.

Eranziel commented:
Hm, I merged this locally, not sure why Github didn't recognize that.

Thanks for the PR! I did a little bit of tidy up after merging, just FYI.
